hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -1641,7 +1641,7 @@ void optimizeARM(Vunit& unit, const Abi& abi, bool regalloc) {
   simplify(unit);
 
   lowerForARM(unit);
-
+  reuseImmq(unit);
```

Comment:
You are right. I am obviously trying to force my ideas into your PR... Sorry for that.
So please put this pass after `foldImms()`, then I am fine.
I'll come up with a PR, which implements my ideas, once this PR lands (and when I'm sure that my idea improves performance in at least once benchmark).

Just one question regarding the naming: why not calling your pass `reuseImms()` (similar to `foldImms()` and `optimizeCopies()` and `annotateSFUses()` and `optimizeJmps()`)?